src/null-safety/index.md

```diff
@@ -1,39 +1,257 @@
 ---
 title: Null safety
-description: Watch this space for news about null safety
-toc: false
+description: Information about Dart's upcoming null safety feature
 ---
 
-Watch this page for updates about null safety.
-In the meantime, check out these resources:
+Sound null safety is coming to the Dart language!
+When you opt into null safety,
+types in your code are non-nullable by default, meaning that
+values can’t be null _unless you say they can be._
+With null safety, your **runtime** null-dereference errors
+turn into **edit-time** analysis errors.
+The compiler can optimize away internal null checks,
+enabling apps to be faster, smaller, and more reliable.
 
-* [Null-safe playground: nullsafety.dartpad.dev][nullsafety.dartpad.dev]
+You can practice using null safety in the web app
+[DartPad with Null Safety.][nullsafety.dartpad.dev]
+
+{{ site.alert.important }}
+  **Don't start migrating large projects yet.**
+  Do test the feature and [give us feedback.][]
+{{ site.alert.end }}
+
+[give us feedback.]: https://github.com/dart-lang/sdk/issues/new?title=Null%20safety%20feedback:%20[issue%20summary]&labels=NNBD&body=Describe%20the%20issue%20or%20potential%20improvement%20in%20detail%20here
+
+If you've used Kotlin, TypeScript, or C#,
+then some of the null-safety syntax might look familiar.
+That's by design: the Dart language aims to be unsurprising.
+
+
+## Creating and using variables (?, !, late)
+
+Use `?`, `!`, and `late` to tweak how the Dart analyzer treats your code.
+
+Here’s an example of declaring a **non-nullable integer variable**
+(assuming you’ve opted into null safety):
+
+```dart
+int aRealInt; // Can't be used until it's initialized to an integer.
+```
+
+If the variable _can_ have the value `null`,
+**add `?`** to its type declaration:
+
+```dart
+int? aNullableInt = null;
+```
+
+When using a value that you know isn’t null but that has a nullable type,
+**add `!`** to tell the Dart analyzer that the value isn’t null:
+
+```dart
+int? a = 2;
+int b = a!; // `a!` is an int (or throws if `a` is null).
+```
+
+If you know that a non-nullable variable will be
+initialized to a non-null value before it's used,
+but the Dart analyzer thinks otherwise,
+**insert `late`** before the variable's type:
+
+```dart
+late int aRealInt;
+```
+
+{% comment %}
+PENDING: Uncomment this once we're ready to offer more guidance.
+
+{{ site.alert.tip }}
+  Avoid using `late final` unless you really need it.
+  For example, prefer using an [initializer list][]
+  to initialize instance variables.
+{{ site.alert.end }}
+
+[initializer list]: /guides/language/language-tour#initializer-list
+{% endcomment %}
+
+
```

Comment:
Consider mentioning the "null aware operator" (`?.`):

```dart
  double? d;  
  print(d?.floor());
```

Developers will see it mentioned in error messages:

```dart
  double? d;  
  print(d.floor());
```
=>

```
 Error: Method 'floor' cannot be called on 'double?' because it is potentially null.
Try calling using ?. instead.
  print(d.floor());
          ^^^^^
```

Reply:
OK, since this isn't really an analysis error, I've put this in a new section called **Other common code changes.**

Reply:
I would like to entirely do away with the section header "Fixing analysis errors". That titles sounds like the developer is in some battle with the analyzer, which hopefully really isn't the case :-). I really like the content in the section, just not the title. Maybe something like "Working with nullable variable safely"?

Also, I think that section needs to be moved up and be right after "Creating and using" (it's all another form of using), and it's stuff that is far more important that developers understand early than the details of list, sets, and maps.

Reply:
Kathy, for an example of what I mean by this being part of "Creating and using" try to ready through the blog post draft, which has this content together with declaring.